python/pyspark/sql/tests/test_types.py

```diff
@@ -196,6 +196,12 @@ def test_infer_nested_schema(self):
         df = self.spark.createDataFrame(nestedRdd2)
         self.assertEqual(Row(f1=[[1, 2], [2, 3]], f2=[1, 2]), df.collect()[0])
 
+        with self.sql_conf({"spark.sql.pyspark.inferNestedStructByMap": False}):
```

Comment:
Can you create a separate test? We should also add a comment, see "Pull Request" in https://spark.apache.org/contributing.html